### PR TITLE
network: Delete conflicting route

### LINKED ISSUE #175: Route table not setup properly for PTP CNI plugin
For full details

https://github.com/containers/virtcontainers/issues/500

### PR DESCRIPTION
**Pull request by amshinde**
If a route with destination exists, delete that route and
add the new route for updated information.

Fixes #175

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

**Review by mcastelino — APPROVED** [review]
@

**Comment by mcastelino** [comment]
@amshinde this actually only partially solves the issue. We may still have routes that should not exist. We can handle that separately. i.e non conflicting interface routes

**Comment by amshinde** [comment]
@mcastelino Yes, this is just to fix the conflicting routes.

**Review by sboeuf — APPROVED** [review]

**Comment by sboeuf** [comment]
I thought the `RouteDel()` was going to fail because the `RouteReplace()` does not work in that case, but I am glad to see this works!
LGTM

File: network.go [hunk 1]
```diff
@@ -193,9 +193,12 @@ func setupRoutes(netHandle *netlink.Handle, routes *[]hyper.Route) error {
 				route.Dest += "/" + netMask
 			}
 
-			if routeDestExist(linkAttrs.Index, initRouteList, route.Dest) {
-				agentLog.WithField("route-destination", route.Dest).Info("Route destination already exists, skipping")
-				continue
+			existingRoute := findExistingDestRoute(linkAttrs.Index, initRouteList, route.Dest)
+			if existingRoute != nil {
+				agentLog.WithField("route-destination", route.Dest).Info("Route destination already exists, deleting")
```
Comment (jodh-intel) [inline network.go:198]: Nice! :smile:

**Comment by jodh-intel** [comment]
lgtm

[Approved with PullApprove](https://pullapprove.com/clearcontainers/agent/pull-request/176/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/agent)

**Pull request merged**